From cf028d2c7afaccd920c5e4da91e9f7e706241cf6 Mon Sep 17 00:00:00 2001
From: Justin Bronder <jsbronder@gmail.com>
Date: Mon, 21 Mar 2011 17:28:37 -0400
Subject: [PATCH] stage:  add gentoo patches

---
 simulator_stage/stage/Makefile.stage |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/simulator_stage/stage/Makefile.stage b/simulator_stage/stage/Makefile.stage
index e9c070d..01d1391 100644
--- a/simulator_stage/stage/Makefile.stage
+++ b/simulator_stage/stage/Makefile.stage
@@ -7,6 +7,8 @@ TARBALL_URL = http://pr.willowgarage.com/downloads/$(FILENAME)
 SOURCE_DIR = build/Stage-3.2.2-Source
 UNPACK_CMD = tar xzf
 MD5SUM_FILE = $(FILENAME).md5sum
+GENTOO_PATCHES = $(wildcard gentoo-patches/*.patch)
+
 include $(shell rospack find mk)/download_unpack_build.mk
 
 ifeq ($(ROS_DEBBUILD),1)
@@ -18,6 +20,9 @@ endif
 
 installed: wiped $(SOURCE_DIR)/unpacked Makefile.stage
 	mkdir -p $(SOURCE_DIR)/build
+	for p in $(GENTOO_PATCHES); do \
+		patch -p1 -d $(SOURCE_DIR) < $$p; \
+	done
 	cd $(SOURCE_DIR)/build && cmake -D CMAKE_INSTALL_PREFIX=$(CMAKE_INSTALL_PREFIX) ..
 	cd $(SOURCE_DIR)/build && make DESTDIR=$(DESTDIR) $(ROS_PARALLEL_JOBS) install
 	touch installed
-- 
1.7.3.4

