From: Justin Bronder <jsbronder@gmail.com>
Date: Tue, 21 Dec 2010 16:52:22 -0500
Subject: [PATCH] makefile: install man pages

---
 Makefile |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/Makefile b/Makefile
index 207d295..9ad0ab8 100644
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,7 @@ dpmpar_.o fdjac2_.o hybrj1_.o lmder_.o  lmpar_.o  qform_.o  r1mpyq_.o covar_.o
 
 # target dir for install
 DESTDIR ?= /usr/local
+MANDIR ?= ${DESTDIR}/share/man
 #
 #  Static library target
 #
@@ -37,6 +38,10 @@ install: libminpack.a
 	chmod 644 ${DESTDIR}/include/minpack.h
 	cp cminpack.h ${DESTDIR}/include
 	chmod 644 ${DESTDIR}/include/cminpack.h
+	install -d ${MANDIR}/man3
+	for f in doc/*.3; do \
+		install -m 644 $$f ${MANDIR}/man3/; \
+	done
 
 clean:
 	rm -f *.o libminpack.a *~ #*#
-- 
